fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -350,17 +355,46 @@ public CompletableFuture<Boolean> markIndexReadable(boolean markReadablePlease)
         if (!markReadablePlease) {
             return AsyncUtil.READY_FALSE; // they didn't say please..
         }
-        AtomicBoolean allReadable = new AtomicBoolean(true);
-        return getRunner().runAsync(context -> openRecordStore(context)
-                .thenCompose(store -> forEachTargetIndex(index ->
-                        store.markIndexReadable(index).thenApply(built -> {
-                            if (!built) {
-                                allReadable.set(false);
-                            }
-                            return null;
-                        }))
-                        .thenApply(ignore -> allReadable.get())
-                ), common.indexLogMessageKeyValues("IndexingBase::markReadable"));
+
+        AtomicReference<RecordIndexUniquenessViolation> uniquenessViolationExRef = new AtomicReference<>();
+        AtomicBoolean anythingChanged = new AtomicBoolean(false);
+
+        // Mark each index readable in its own (retriable, parallel) transaction. If one target fails to become
```

Comment:
That is quite possible that it would be too complex, and not worth it. I just wanted to bring it up as something to consider, if it didn't add much to the complexity.